Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSecurityEventFirewall.yaml

```diff
@@ -0,0 +1,303 @@
+              '\x0d\x0a  <Data Name="SourcePort">'SrcPortNumber:int'</Data>'
+              '\x0d\x0a  <Data Name="DestAddress">'DstIpAddr:string'</Data>'
+              '\x0d\x0a  <Data Name="DestPort">'DstPortNumber:int'</Data>'
+              '\x0d\x0a  <Data Name="Protocol">'Protocol:int'</Data>'
+              '\x0d\x0a  <Data Name="FilterRTID">'NetworkRuleNumber:int'</Data>'
+              '\x0d\x0a  <Data Name="LayerName">'LayerCode'</Data>'
+              '\x0d\x0a  <Data Name="LayerRTID">'LayerRTID'</Data>'
+              '\x0d\x0a  <Data Name="RemoteUserID">'RemoteUserID:string'</Data>'
+              '\x0d\x0a  <Data Name="RemoteMachineID">'RemoteMachineID:string'</Data>'*
+              | project-away EventData
+              | extend temp_isSrcMatch=has_any_ipv4_prefix(SrcIpAddr,src_or_any)
+               , temp_isDstMatch=has_any_ipv4_prefix(DstIpAddr,dst_or_any)
+              | extend ASimMatchingIpAddr = case(
+              array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-" // match not requested: probably most common case
+              , (temp_isSrcMatch and temp_isDstMatch), "Both" // has to be checked before the individual 
+              , temp_isSrcMatch, "SrcIpAddr"
+              , temp_isDstMatch, "DstIpAddr"
+              , "No match"
+              )
+             | where ASimMatchingIpAddr != "No match"
+             | project-away temp_*
+          ;
+      union SecurityEvent_5154_5155_5158_5159, SecurityEvent_5156_5157, SecurityEvent_5152
+      | lookup Directions on DirectionCode
+      | project-rename DvcHostname = Computer
+      | extend
+            SrcAppName = iff(isOutBound, Application, ""),
+            DstAppName = iff(not(isOutBound), Application, ""),
+            SrcDvcId = iff(isOutBound, RemoteMachineID, ""),
+            DstDvcId = iff(not(isOutBound), RemoteMachineID, ""),
+            SrcProcessId =  iff(isOutBound, tostring(ProcessId), ""),
+            DstProcessId =  iff(not(isOutBound), tostring(ProcessId), ""),
+            DstUserId = iff(isOutBound, RemoteUserID, ""),
+            SrcUserId = iff(not(isOutBound), RemoteUserID, ""),
+            DstHostname = iff(isOutBound, "", DvcHostname),
+            SrcHostname = iff(isOutBound, DvcHostname, "")
+          | project-away Application, RemoteMachineID, ProcessId, RemoteUserID
+            // *************** Postfilterring *****************************************************************
+          | where  (isnull(dstportnumber) or DstPortNumber == dstportnumber )
+          // *************** / Postfilterring *****************************************************************
+      };
+  // Main query -> outputs both schemas as one normalized table
+  WindowsFirewall_SecurityEvent  (starttime, endtime, srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix, dstportnumber, hostname_has_any, dvcaction, eventresult, disabled)
```

Comment:
can we merge this in previous block? we would not need this line then

Reply:
done